OfficeDev/Excel-Custom-Functions
# CustomFunctions.associate()

The method to connect the implementation of the custom function to the id specified in the metadata has changed. Use the CustomFunctions.associate(id: string, func: Function) method provided by the custom functions runtime instead of CustomFunctionMappings object.

```
CustomFunctions.associate("add", add);
```
instead of
```
CustomFunctionMappings.ADD = add;
```
---
**Hosts/Platforms tested:**
- [x] Excel for Windows
- [x] Excel Online - Windows (Edge)
- [x] Excel Online - Mac (Chrome)

**Validation performed:**

Excel Online:
1. npm install
2. npm run build
3. npm run dev-server
4. Open in browser: https://www.office.com
5. Create new Excel document
6. Insert add-in via manifest upload.
7. Type in cell: "=CONTOSO.ADD(1, 2)". Validate that result is shown: 3
8. Press F12 to open browser dev tools
9. Ctrl+P and open "index.win32.bundle"
10. Set breakpoint inside "add()" method in code.
11. Type in cell: "=CONTOSO.ADD(3, 4)".
12. Verify hit breakpoint.

Excel for Windows:
1. npm install
2. npm run build
3. start npm run dev-server
4. npm run sideload
5. In Excel, type in cell: "=CONTOSO.ADD(1, 2)". Validate that result is shown: 3
